### DISCUSSION TITLE
Why is the rectangle not curved on the globe?

### DISCUSSION THREAD

**Discussion by tangzijun** [maplibre/maplibre-gl-js#4887] [Q&A]
When the latitude and longitude are in the following manner, on the globe, if the rectangle is not curved, what measures can be taken?

In addition, I use a custom layer to render the rectangle

```
const leftTop = mapgl.MercatorCoordinate.fromLngLat({ lng: -30, lat: 30 });
const rightTop = mapgl.MercatorCoordinate.fromLngLat({ lng: 30, lat: 30 });
const rightBottom = mapgl.MercatorCoordinate.fromLngLat({ lng: 30, lat: -30 });
const leftBottom = mapgl.MercatorCoordinate.fromLngLat({ lng: -30, lat: -30 });
```

![image](https://github.com/user-attachments/assets/900e7238-e03c-46fc-b017-8f26e88dbd95)
![image](https://github.com/user-attachments/assets/b6f01f3a-087d-44c9-a04c-d0c4a592bf45)

**Comment by zerda-ocm** [comment]
The lines between the edges need some more points in between. Also known as subdivision.

**Comment by tangzijun** [comment]
> The lines between the edges need some more points in between. Also known as subdivision.The lines between the edges need some more points in between. Also known as subdivision.边之间的线之间需要更多的点。也称为细分。

Is it cut into tiles?

**Comment by zerda-ocm** [comment]
No, it is not cut into tiles. If you want the square to follow the curvature of the globe, you need to give the renderer some more points to work with it. Think of it like a steel rod you want to place on the surface of basketball. You would definitely have some trouble. Now try the same with a chain link.

**Comment by tangzijun** [comment]
> No, it is not cut into tiles. If you want the square to follow the curvature of the globe, you need to give the renderer some more points to work with it. Think of it like a steel rod you want to place on the surface of basketball. You would definitely have some trouble. Now try the same with a chain link.

Sorry, I think I understand what you mean, but I don't know how to implement it. I ultimately need to display a large image that fits on the sphere, but I'm not sure if this is possible? Are there any code examples that can help me?

**Comment by zerda-ocm** [comment]
This should do it:
https://jsbin.com/yetujagopu/1/edit?html,console,output

**Comment by tangzijun** [comment]
> This should do it: https://jsbin.com/yetujagopu/1/edit?html,console,output

I tested it and the raster layer is indeed fine. I need to implement the mapping in a custom layer because I need to write glsl to change the texture.

I am now looking at the raster implementation source code, but I am not sure I can figure it out. Is there an example of implementing spherical mapping in a custom layer?

![image](https://github.com/user-attachments/assets/75690e94-52f3-4283-adf4-9ec202c58211)
